corehq/apps/app_manager/models.py

```diff
@@ -4474,13 +4474,18 @@ def make_build(self, comment=None, user_id=None):
 
     def check_build_dependencies(self, new_build):
         """
-        This method reports whether the app dependencies have been removed.
+        Reports whether the app dependencies have been removed.
         """
 
         def has_dependencies(build):
-            return bool(build['profile']['features']['dependencies'])
+            return bool(
+                build.get('profile', {}).get('features', {}).get('dependencies')
+            )
 
         latest_build = get_latest_build_doc(self.domain, self.id)
+        if not latest_build:
```

Comment:
When would there not be a latest build?

Reply:
The "latest_build" here refers to the previous build, as the new build has not been made at this point. So the "latest_build" could be `None` if it's the first build.

Reply:
Ah I see, thanks for the clarification

Reply:
nit: to make it slightly better, you could consider naming this last_build.

Reply:
https://github.com/dimagi/commcare-hq/pull/35455/commits/c2ef0d0572be34cf0e3418e977281296dc9ae455